### PR TITLE
Progress logs

### PR DESCRIPTION
**Pull request by ScienceMaster7**
What I did:

- I added the property dateTracker to the habit object.
- I set up the logic to store the time the user submitted, with the current date.
- I created and linked the page Progress logs.
- I created cards for each habit and set up how the progress should be displayed on them.

I still need to add some logic to also display the days the user was inactive.

Constructive feedback is appreciated !

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -2,14 +2,13 @@
 <html lang="en">
   <head>
     <meta charset="utf-8" />
-    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <link rel="icon" href="./favicon.png" />
```
Comment (johnxgaehner) [inline public/index.html:5]: cool favicon, looks nice!

File: src/pages/ProgressLogs.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import Footer from "../components/Footer";
+import StreakItems from "../components/StreakItems";
+import "./ProgressLogs.css";
+
+export default function ProgressLogs() {
+  const habits = JSON.parse(localStorage.getItem("habits"));
+  let dateAndTime;
+  if (habits) {
```
Comment (johnxgaehner) [inline src/pages/ProgressLogs.js:8]: you could use this to return a feedback if there are no habits yet like you've done in your `home` page.

File: src/App.js [hunk 3]
```diff
@@ -32,6 +33,9 @@ function App() {
           <Route path="/create">
             <CreateHabit />
           </Route>
+          <Route path="/progressLogs">
```
Comment (johnxgaehner) [inline src/App.js:36]: I did some research about the casing convention in urls and the result was, that you should go with hivens and avoid CamelCase.

The best thing for search engine optimization is to use hyphens, then each "word" in the URL is considered a "term" by the search engines. Having no spaces between the words means that the entire string is going to be treated as a single word.

File: src/pages/CreateHabit.js [hunk 4]
```diff
@@ -15,6 +18,44 @@ export default function CreateHabit() {
...
+          timeToday: 0,
+        },
+        {
+          date: "16.08.2021",
+          timeToday: 0,
+        },
+        {
+          date: today,
+          timeToday: 0,
+        },
```
Comment (johnxgaehner) [inline src/pages/CreateHabit.js:57]: this is probably only for development, right? shouldn't be in the pull request. really confused be while trying out the app to understand what the progress log is doing.

File: src/pages/CreateHabit.js [hunk 5]
```diff
@@ -15,6 +18,44 @@ export default function CreateHabit() {
...
+      ],
     };
 
     const habbitList = localStorage.getItem("habits");
```
Comment (johnxgaehner) [inline src/pages/CreateHabit.js:61]: in line 69 you're checking if there is a habit with the real equal name but if you write it once with big and the other time with small letters, it's valid.

Screenshot 2021-08-18 at 09 13 34

File: src/pages/ProgressLogs.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    dateAndTime = habits.map((habit, index) => {
+      console.log(habit.dateTracker[0].date);
```
Comment (johnxgaehner) [inline src/pages/ProgressLogs.js:10]: no console.log in pr. :)))

File: src/styles/colors.css [hunk 7]
```diff
@@ -5,4 +5,5 @@
   --sencodary-text-color: #000000;
```
Comment (johnxgaehner) [inline src/styles/colors.css:5]: little typo.

**Review by johnxgaehner — COMMENTED** [review]
Hey Bene! really nice.
especially having these helper functions is great and structured.

left you some comments.

also while trying your app it was a bit confusing to have the menu and again a navbar with your home in the bottom. maybe it's better structured if you decide for one to have everything in one place? I personally really like the menu. :)

File: src/pages/ProgressLogs.css [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+.streak__item {
+  background-image: linear-gradient(
+    to bottom right,
+    var(--streak-gradient-color)
+  );
+  height: 2rem;
+  width: 2rem;
+  margin: 0 0 1rem 0;
```
Comment (me-jterhorst) [inline src/pages/ProgressLogs.css:39]: Wouldn't you be better off by just writing margin-bottom: 1rem?

**Pull request merged**